src/app/components/Table/GoToFirstPageLink.tsx

```diff
@@ -0,0 +1,33 @@
+import { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useSearchParams, useHref } from 'react-router-dom'
+import Link from '@mui/material/Link'
+
+export const GoToFirstPageLink: FC = () => {
+  const { t } = useTranslation()
+  const [searchParams] = useSearchParams()
+  // We have no sure way to know the number of the page parameter to remove.
+  // The best we can do is to test the trivial option.
+  // If it's not there, then we can't really do anything.
+  const wantedParamName = searchParams.has('page') ? 'page' : undefined
+  const label = t('errors.loadFirstPage')
+  const newSearchParams = new URLSearchParams(searchParams)
+  if (wantedParamName) {
+    newSearchParams.delete(wantedParamName)
+  }
+  const href = useHref({
+    search: newSearchParams.toString(),
+  })
+
+  // Ideally, we should use a router link here, but even if we change the URL without a reload,
+  // I can't get the error boundary to recover from the error state.
+  // I tried to implement recovery, but it added too much complexity,
+  // so it's easier to just reload in these cases.
```

Comment:
I don't understand

- https://7ffc492f.oasis-explorer.pages.dev/mainnet/sapphire/address/0x46b983CF8f03db357Cf10ADE79a65eC76DE0BF49?page=100x
- click Token Transfers
- click Transactions
- working state

is this no-reload recovered state you are trying to get?

Reply:
The above process works by switching tabs, which means unmounting and re-rendering the whole DOM of the tab. So the `ErrorBoundary` component is fully destroyed and re-rendered. 

However, what we need here would be that the `ErrorBoundary` component stays in place, we just navigate _inside_ the same tab (to a different page). In order to actually need the first page, the Error Boundary component would need to switch back from the "error" state to the "normal" state, which doesn't happen automatically.

In needs a manual `setState()` for recovering,  which is hard to channel through, but even if we do it (which I did as an experiment), we find that the data around the Error Boundary is somewhat weird, and the updated URL is not reflected right away, so we just end up running into the same exception again...

So apparently the solution is to umount and rerender the same component outside the error boundary.

But that's hard to do selectively, so it's easier to just to a full refresh.

Please note that to get to an invalid / non-existent page, the user probably needs to do some manual URL hacking. I don't think it's unacceptable behavior on our part that he gets a full reload as a result.

Reply:
hm, alright

Can you add more of this context to the comment, and add a component-level comment this component is only "best effort", and is only rarely triggered by user changing URL

Reply:
OK done